kore/src/Kore/Exec.hs

```diff
@@ -648,13 +644,16 @@ initializeProver definitionModule specModule within =
         -- This assertion should come before simplifiying the claims,
         -- since simplification should remove all trivial claims.
         assertSomeClaims specClaims
+        let firstClaim = snd . head $ specClaims
         simplifiedSpecClaims <-
             mapM (mapMSecond simplifyToList) specClaims
         specAxioms <- Profiler.initialization "simplifyRuleOnSecond"
             $ traverse simplifyRuleOnSecond (concat simplifiedSpecClaims)
-        let
-            axioms = coerce <$> rewriteRules
-            claims = fmap makeClaim specAxioms
+        let claims = fmap makeClaim specAxioms
+            axioms =
+                Goal.fromRulePattern (Goal.goalToRule firstClaim)
```

Comment:
Why are we inserting the first claim into the list of axioms?

Reply:
This doesn't insert the claim, we just need to get the axiom type here which is determined by the claim type. If we revert the generalization of `Exec.prove` I think this awkward way of getting the axiom type at start-up won't be needed anymore.